File: frontend/src/types/expense.ts
export enum CategoryEnum {
  FOOD = "Food",
  HOUSING = "Housing",
  UTILITIES = "Utilities",
  ENTERTAINMENT = "Entertainment",
  SHOPPING = "Shopping",
  HEALTHCARE = "Health Care",
  TRAVEL = "Travel",
  OTHER = "Other",
}

export interface Expense {
  id: number;
  amount: number;
  description?: string;
  category: CategoryEnum;
  date: string;
}

export interface ExpenseCreate {
  amount: number;
  description?: string;
  category: CategoryEnum;
}

export interface ExpenseUpdate {
  amount: number;
  description?: string;
  category: CategoryEnum;
}

export interface TotalsResponse {
  total: number;
  breakdown: Record<string, number>;
}

File: backend/app/schemas/expense.py
from pydantic import BaseModel,Field
from typing import Optional
from datetime import datetime
from .category_enum import CategoryEnum

class ExpenseBase(BaseModel):
    amount:float= Field(...,gt=0)
    description:str|None=None
    category:CategoryEnum

class ExpenseCreate(ExpenseBase):
    pass

class ExpenseUpdate(ExpenseBase):
    pass

class ExpenseOut(ExpenseBase):
    id:int
    date:datetime
    class Config:
        orm_mode = True

